sdk/iOS/OoyalaSkinSDK/OOUpNextManager.m

```diff
@@ -82,7 +84,37 @@ - (void)goToNextVideo {
   // if upnext has not been dismissed and there is a next video, play the next video.
   LOG(@"Going to next video based on Up Next");
   if (!self.isDismissed && self.nextVideo) {
-    [self.player setEmbedCode:self.nextVideo[embedCodeKey]];
+    
+#warning: old API. Remove when SDK version > 4.46.0_GA. If you need to use old API, uncomment -playForJustChangedItem in -bridgeCurrentItemChangedNotification: of 'OOSkinPlayerObserver'
+    //[self.player setEmbedCode:self.nextVideo[embedCodeKey]];
+    
+    //new API from OOyalaSDK. Available > 4.46.0_GA
+    [self.player setEmbedCode:self.nextVideo[embedCodeKey] withCallback:^(OOOoyalaError *error) {
+      LOG(@"goToNextVideo setEmbedCode got callback");
+      if (error) {
+        LOG(@"❌ Is setEmbedCode is NOT successfull. Error: %@", error.debugDescription);
+      }
+    }];
+    
```

Comment:
I see that we make `setEmbedCode:`  call and then we continue execution not after callback is returned. Is it expected?
What is `currentItemChangedCallback`? Why we set it after calling `setEmbedCode:`?

Reply:
Yes, block doesn't rely on on async setEmbedCode:withCallback: , block waits for event that can't be pass back to  setEmbedCode:withCallback:. This event - it's moment when KVO observes that state of AVPlayer changed to `AVPlayerItemStatusReadyToPlay`
Chain of events:
1 - OOOyalaPlayer sets embed code
2 - OOOyalaPlayer set URL with streams to OOBaseStreamPlayer
3 - OOBaseStreamPlayer pass URL to native AVPlayer
4 - KVO observes that AVPlayer utilize URL with success and ready to play
(please, see attachments) -> callback can be triggered

Anyway I'll define `currentItemChangedCallback` before setEmbedCode: to avoid situations when setEmbedCode: interrupted and setEmbedCode:withCallback: with error is returned

Reply:
Changed